src/mame/tandberg/tdv2100_disp_logic.cpp

```diff
@@ -0,0 +1,1029 @@
+	PALETTE(mconfig, m_palette, palette_device::MONOCHROME_HIGHLIGHT);
+
+	/* Sound */
+	SPEAKER(mconfig, "mono").front_center();
+	BEEP(mconfig, m_beep, 2000);
+	m_beep->add_route(ALL_OUTPUTS, "mono", 1.0);
+
+	RS232_PORT(mconfig, m_rs232, default_rs232_devices, nullptr);
+	m_rs232->rxd_handler().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_rxd_w));
+	m_rs232->dcd_handler().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_dcd_w));
+	m_rs232->dsr_handler().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_dsr_w));
+	m_rs232->cts_handler().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_cts_w));
+	m_rs232->ri_handler().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_ri_w));
+
+	AY51013(mconfig, m_uart);
+	m_uart->write_so_callback().set(FUNC(tandberg_tdv2100_disp_logic_device::rs232_txd_w));
+	m_uart->write_dav_callback().set(FUNC(tandberg_tdv2100_disp_logic_device::uart_rx));
+	m_uart->write_dav_callback().append(FUNC(tandberg_tdv2100_disp_logic_device::check_rs232_rx_error));
+	m_uart->write_tbmt_callback().set(FUNC(tandberg_tdv2100_disp_logic_device::uart_tx));
+	m_uart->write_pe_callback().set(FUNC(tandberg_tdv2100_disp_logic_device::check_rs232_rx_error));
+
+	CLOCK(mconfig, m_uart_clock, 1);
+	// NOTE: Frequency set with the rest of the UART settings
+	m_uart_clock->signal_handler().set(m_uart, FUNC(ay51013_device::write_rcp));
+	m_uart_clock->signal_handler().append(m_uart, FUNC(ay51013_device::write_tcp));
+}
+
+static INPUT_PORTS_START( tdv2115l )
+
+	PORT_START("sw_invert_video")
+		PORT_CONFNAME(0x1, 0x1, "INVERSE VIDEO")
+			PORT_CONFSETTING(0x1, DEF_STR( Off ))
+			PORT_CONFSETTING(0x0, DEF_STR( On ))
+
+	PORT_START("sw_rs232_baud")
+		PORT_CONFNAME(0x7, 0x6, "SPEED SELECT (Baud-rate)")                         PORT_CHANGED_MEMBER(DEVICE_SELF, tandberg_tdv2100_disp_logic_device, uart_changed, 0)
+			PORT_CONFSETTING(0x0, "0: 110")
+			PORT_CONFSETTING(0x1, "1: 300")
+			PORT_CONFSETTING(0x2, "2: 600")
+			PORT_CONFSETTING(0x3, "3: 1200")
+			PORT_CONFSETTING(0x4, "4: 2400")
+			PORT_CONFSETTING(0x5, "5: 4800")
+			PORT_CONFSETTING(0x6, "6: 9600")
+			PORT_CONFSETTING(0x7, "7: 19200")
+
+	PORT_START("sw_rs232_settings")
+		PORT_CONFNAME(0x01, 0x01, "NO PARITY  (No: Enable parity)")                 PORT_CHANGED_MEMBER(DEVICE_SELF, tandberg_tdv2100_disp_logic_device, uart_changed, 0)
+			PORT_CONFSETTING(0x00, DEF_STR( No ))
+			PORT_CONFSETTING(0x01, DEF_STR( Yes ))
+		PORT_CONFNAME(0x02, 0x02, "EVEN PARITY (No: Odd parity, when enabled)")     PORT_CHANGED_MEMBER(DEVICE_SELF, tandberg_tdv2100_disp_logic_device, uart_changed, 0)
+			PORT_CONFSETTING(0x00, DEF_STR( No ))
+			PORT_CONFSETTING(0x02, DEF_STR( Yes ))
+		PORT_CONFNAME(0x04, 0x04, "TWO STOP BITS (No: One stop bit)")               PORT_CHANGED_MEMBER(DEVICE_SELF, tandberg_tdv2100_disp_logic_device, uart_changed, 0)
+			PORT_CONFSETTING(0x00, DEF_STR( No ))
+			PORT_CONFSETTING(0x04, DEF_STR( Yes ))
```

Comment:
This is a question about if we should be accurate to the hardware or accurate to the user-friendliness. I had the impression that Mame opted to be more on the side of being as much in line with the hardware as possible.

The names used here are literally what's printed on the switch-panel of the machine, including the "Yes/No" options. Only the content of the parenthesis is added for clarity. If I use more descriptive terms, the driver gets more user-friendly for the average user, but less preservative of the actual machine.

Reply:
Could you add a comment above explaining the wording is what's printed on the switch-panel so people know you didn't choose it?

Reply:
That sounds like a good idea, added a note about this

Reply:
In general, we tend to prefer to make the setting labels as clear as possible in MAME.  You need to remember that the majority of people using MAME are not going to have the original manual at hand.  In many cases, we use standardised names for setting labels across systems in MAME rather than the wording used in original manuals or test modes (we have constants for standardised names for many settings that are common across arcade games in particular).

Here’s a somewhat extreme example: https://github.com/mamedev/mame/blob/mame0261/src/mame/bmc/bmcpokr.cpp#L834

The system displays the settings in Chinese, and attempting a literal translation would be an exercise in futility.  The actual displayed text is preserved in comments.

Reply:
Allright, if so be.